Add docs (#294)
CQCL/hugr-qir: docs/guppy-for-h-series/index.md

# Guppy for H-Series

`hugr-qir` is aimed at Guppy programs that compile through HUGR and then target Quantinuum's H-Series systems via QIR.

> That workflow is already usable today, but it supports a constrained subset of Guppy. This section breaks down the current support envelope and the main modeling constraints to keep in mind when writing programs for H-Series targets.

Reviewer: ```suggestion
That workflow is already usable today, but it supports only a subset of Guppy. This section summarizes the current limitations to keep in mind when writing programs for H-Series targets.
```
